sdk/communication/azure-communication-administration/src/test/java/com/azure/communication/administration/PhoneNumberAsyncClientPlaybackTest.java

```diff
@@ -0,0 +1,361 @@
+
+    @Test()
+    public void purchaseSearchWithResponse() {
+        Mono<Response<Void>> mono = this.getClient().purchaseSearchWithResponse(SEARCH_ID_TO_PURCHASE, Context.NONE);
+
+        Response<Void> response = mono.block();
+        assertEquals(202, response.getStatusCode());
+    }
+
+    @Test()
+    public void cancelSearch() {
+        Mono<Void> mono = this.getClient().cancelSearch(SEARCH_ID_TO_CANCEL);
+        mono.block();
+    }
+
+    @Test()
+    public void cancelSearchWithResponse() {
+        Mono<Response<Void>> mono = this.getClient().cancelSearchWithResponse(SEARCH_ID_TO_CANCEL, Context.NONE);
+
+        Response<Void> response = mono.block();
+        assertEquals(202, response.getStatusCode());
+    }
+
+    @Test()
+    public void configureNumber() {
+        PhoneNumber number = new PhoneNumber(PHONENUMBER_TO_CONFIGURE);
+        PstnConfiguration pstnConfiguration = new PstnConfiguration();
+        pstnConfiguration.setApplicationId("ApplicationId");
+        pstnConfiguration.setAzurePstnTargetId("AzurePstnTargetId");
+        pstnConfiguration.setCallbackUrl("https://callbackurl");
+
+        Mono<Void> mono = this.getClient().configureNumber(number, pstnConfiguration);
+
+        mono.block();
+    }
+
+    @Test()
+    public void configureNumberWithResponse() {
+        PhoneNumber number = new PhoneNumber(PHONENUMBER_TO_CONFIGURE);
+        PstnConfiguration pstnConfiguration = new PstnConfiguration();
+        pstnConfiguration.setApplicationId("ApplicationId");
+        pstnConfiguration.setAzurePstnTargetId("AzurePstnTargetId");
+        pstnConfiguration.setCallbackUrl("https://callbackurl");
+
+        Mono<Response<Void>> mono = this.getClient().configureNumberWithResponse(number, pstnConfiguration, Context.NONE);
+
+        Response<Void> response = mono.block();
+        assertEquals(202, response.getStatusCode());
+    }
+
+    @Test()
+    public void getNumberConfiguration() {
+        PhoneNumber number = new PhoneNumber(PHONENUMBER_TO_GET_CONFIG);
+
+        Mono<NumberConfigurationResponse> mono = this.getClient().getNumberConfiguration(number);
+
+        assertEquals("ApplicationId", mono.block().getPstnConfiguration().getApplicationId());
+    }
+
+    @Test()
+    public void getNumberConfigurationWithResponse() {
+        PhoneNumber number = new PhoneNumber(PHONENUMBER_TO_GET_CONFIG);
+
+        Mono<Response<NumberConfigurationResponse>> mono =
+            this.getClient().getNumberConfigurationWithResponse(number, Context.NONE);
+
+        Response<NumberConfigurationResponse> response = mono.block();
+        assertEquals(200, response.getStatusCode());
+        assertEquals("ApplicationId", response.getValue().getPstnConfiguration().getApplicationId());
+    }
+
+    @Test()
+    public void unconfigureNumber() {
+        PhoneNumber number = new PhoneNumber(PHONENUMBER_TO_UNCONFIGURE);
+
+        Mono<Void> mono = this.getClient().unconfigureNumber(number);
+
+        mono.block();
```

Comment:
Do not block when testing async clients. Use `StepVerifier` instead.

Reply:
Will take a look at StepVerifier.

Reply:
Updated async client tests to use StepVerifier.